libraries/standard/http/test/common.h

```diff
@@ -7,4 +7,5 @@
 #include "private/http_client_internal.h"
 #include "private/http_client_parse.h"
 
-#define assert( x )
+/*#define assert( x ) */
+#include <assert.h>
```

Comment:
I remember @sarenameas telling us to avoid doing this

Reply:
That was only for having 100% branch coverage statistics. However, the advantage of having asserts enabled is prevention of missing out build issues in assert expressions. 
(Also, ideally, we should have written tests for assert expression in unit test code, but it may not be feasible to do that in CMock framework; anyway, if we don't want to do that, we can provide evidence of 100% without asserts in PR descriptions)

Reply:
Have added a configuration to enable/disable asserts in unit tests for flexibility